File: cpp/src/patterns/state/state/NoMoneyState.cc
#include "NoMoneyState.h"
#include "VendingMachine.h"


NoMoneyState::NoMoneyState(VendingMachine *machine) {
    machine_ = machine;
}

void NoMoneyState::Insert() {
    cout << "You inserted the money" << endl;
    machine_->SetState(machine_->GetHasMoney());
}

void NoMoneyState::Eject() {
    cout << "You haven't inserted money" << endl;
}

void NoMoneyState::Bought() {
    cout << "You turned but there's no money" << endl;
}

void NoMoneyState::Dispense() {
    cout << "You need to pay first" << endl;
}

File: Java/src/patterns/state/stateInterface/NoMoneyState.java
package patterns.state.stateInterface;

public class NoMoneyState implements State {
	private VendingMachine machine;
	
	public NoMoneyState(VendingMachine machine) {
		this.machine = machine;
	}
	
	@Override
	public void insert() {
		System.out.println("You inserted the money");
		machine.setState(machine.getHasMoeny());
	}

	@Override
	public void eject() {
		System.out.println("You haven't inserted money");
	}

	@Override
	public void bought() {
		System.out.println("You turned but there's no money");
	}

	@Override
	public void dispense() {
		System.out.println("You need to pay first");
	}
}
